src/playground-code-editor.ts

```diff
@@ -101,9 +111,36 @@ export class PlaygroundCodeEditor extends LitElement {
     playgroundStyles,
   ];
 
-  // Used by tests.
   protected _codemirror?: ReturnType<typeof CodeMirror>;
 
+  get cursorPosition(): Position | undefined {
+    const cm = this._codemirror;
+    if (!cm) return undefined;
+
+    return cm.getCursor('start');
+  }
+
+  get cursorIndex(): number | undefined {
```

Comment:
Similar to above, maybe we should default to 0 if codemirror is uninitialized? Since it will always start there anyway?